src/HTTPRequest.php

```diff
@@ -2,36 +2,68 @@
 
 namespace MeiliSearch;
 
-use GuzzleHttp\Client as GuzzleClient;
-use GuzzleHttp\Exception\BadResponseException;
-use GuzzleHttp\Psr7\Request as GuzzleRequest;
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\Psr17FactoryDiscovery;
 use MeiliSearch\Exceptions\HTTPRequestException;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamFactoryInterface;
 
 class HTTPRequest
 {
+    /**
+     * @var null
+     */
     public $base_url = null;
+
+    /**
+     * @var null
+     */
     public $api_key = null;
-    public $client = null;
+
+    /**
+     * @var array
+     */
     public $headers = [];
 
-    public function __construct($url, $api_key = null)
+    /**
+     * @var ClientInterface
+     */
+    protected $httpClient;
+
+    /**
+     * @var RequestFactoryInterface
+     */
+    private $requestFactory;
+
+    /**
+     * @var StreamFactoryInterface
+     */
+    private $streamFactory;
+
+    public function __construct($url, $api_key = null, ClientInterface $httpClient = null, RequestFactoryInterface $requestFactory = null, StreamFactoryInterface $streamFactory = null)
```

Comment:
This not a constructive criticism but I see that in a lot of places the variable names are snake_case and you have used a mix of snake_cased and camelCased variable names. Maybe we could stick to one practice? I suggest camelCase to be used everywhere.